### PR TITLE
1231 icon delay

### LINKED ISSUE #1231: Toolbar button animation tweaks
## Steps to Reproduce

1. Create thought `a`.
2. Quickly click and release a toolbar button.

## Current Behavior

When the mouse button is quickly released, the dip animation is quickly reversed, showing little change.

In this video, there is a long click (correct) followed by a short click (incorrect).

https://github.com/cybersemics/em/assets/750276/ead547e6-98c8-4efd-b59d-5f8d0af446da

## Expected Behavior

- Complete the full dip animation even if only quickly tapping and releasing.
- Decrease the animation transition duration for a snappier effect.

### PR DESCRIPTION
**Pull request by BlaineOmega**
Fixes #1231 Used timeout option to fix the issue with the icon delay. I removed unused onTapCancel and no-longer-needed onTapUp from the Toolbar.ts component to make the implementation cleaner and easier to understand.

### CODE REVIEW

File: src/components/ToolbarButton.tsx [hunk 1]
```diff
@@ -99,7 +100,7 @@ const ToolbarButtonComponent: FC<DraggableToolbarButtonProps> = ({
       }
     },
     // eslint-disable-next-line react-hooks/exhaustive-deps
-    [longPressTapUp, customize, isButtonExecutable, disabled],
+    [longPressTapUp, customize, isButtonExecutable, disabled, onTapUp],
```
Comment (raineorshine) [inline src/components/ToolbarButton.tsx:103]: We can remove `onTapUp` since it is no longer used.

File: src/components/Toolbar.tsx [hunk 2]
```diff
@@ -94,12 +100,17 @@ const Toolbar: FC<ToolbarProps> = ({ customize, onSelect, selected }) => {
...
-    /**
-     * Update isPressed when the drag is released outside of the toolbar.
-     * */
+    /** Method to handle the release of the mouse. */
     const handleMouseUp = event => {
-      setPressingToolbarId(null)
+      setTimeout(
+        () => {
+          setPressingToolbarId(null)
+          onSelect?.(shortcutById(id))
```
Comment (raineorshine) [inline src/components/Toolbar.tsx:110]: `onSelect` should only be called when selecting a toolbar button, not deselecting.

**Review by raineorshine — CHANGES_REQUESTED** [review]
Works great!

File: src/components/Toolbar.tsx (hunk 2 above)
Comment (BlaineOmega) [reply src/components/Toolbar.tsx]: I added it only because it was there before; removed

**Review by raineorshine — APPROVED** [review]

**Pull request merged**